packages/engine/paima-funnel/src/funnels/evm/funnel.ts

```diff
@@ -0,0 +1,565 @@
+import {
+  ENV,
+  EvmConfig,
+  doLog,
+  initWeb3,
+  logError,
+  timeout,
+  Web3,
+  delay,
+  InternalEventType,
+} from '@paima/utils';
+import type { ChainFunnel, ReadPresyncDataFrom } from '@paima/runtime';
+import type { ChainData, PresyncChainData } from '@paima/sm';
+import { getUngroupedCdeData } from '../../cde/reading.js';
+import { composeChainData, groupCdeData } from '../../utils.js';
+import { BaseFunnel } from '../BaseFunnel.js';
+import type { FunnelSharedData } from '../BaseFunnel.js';
+import {
+  EvmFunnelCacheEntry,
+  EvmFunnelCacheEntryState,
+  RpcCacheEntry,
+  RpcRequestState,
+} from '../FunnelCache.js';
+import type { PoolClient } from 'pg';
+import { FUNNEL_PRESYNC_FINISHED, ConfigNetworkType } from '@paima/utils';
+import { getMultipleBlockData } from '../../reading.js';
+import { getLatestProcessedCdeBlockheight } from '@paima/db';
+
+const GET_BLOCK_NUMBER_TIMEOUT = 5000;
+
+export class EvmFunnel extends BaseFunnel implements ChainFunnel {
```

Comment:
what would work though? `parallelEvmFunnel` maybe? `companionEvmChainFunnel` ? 

I didn't want to name it `sidechainEvmFunnel` since I feel that may be confusing because the word sidechain already means something.

`PairedEvmFunnel` maybe?